### PR TITLE
WIP: Prevent privilege escalation

### PR DESCRIPTION
**Pull request by erikwilson**
The purpose of this change is to prevent privilege escalation when assigning roles for global users, cluster members, or project members.

Issue: https://github.com/rancherlabs/rancher-security/issues/29

### CODE REVIEW

File: pkg/api/server/managementstored/setup.go [hunk 1]
```diff
@@ -424,15 +427,15 @@ func PodSecurityPolicyTemplate(schemas *types.Schemas, management *config.Scaled
...
-}
+// func ClusterRoleTemplateBinding(schemas *types.Schemas, management *config.ScaledContext) {
+// 	schema := schemas.Schema(&managementschema.Version, client.ClusterRoleTemplateBindingType)
+// 	schema.Validator = roletemplatebinding.NewCRTBValidator(management)
+// }
 
-func ProjectRoleTemplateBinding(schemas *types.Schemas, management *config.ScaledContext) {
-	schema := schemas.Schema(&managementschema.Version, client.ProjectRoleTemplateBindingType)
-	schema.Validator = roletemplatebinding.NewPRTBValidator(management)
-}
```
Comment (nathan-jenan-rancher) [inline pkg/api/server/managementstored/setup.go]: Delete commented out code

File: pkg/api/server/managementstored/setup.go [hunk 2]
```diff
@@ -424,15 +427,15 @@ func PodSecurityPolicyTemplate(schemas *types.Schemas, management *config.Scaled
...
+// func ProjectRoleTemplateBinding(schemas *types.Schemas, management *config.ScaledContext) {
+// 	schema := schemas.Schema(&managementschema.Version, client.ProjectRoleTemplateBindingType)
+// 	schema.Validator = roletemplatebinding.NewPRTBValidator(management)
+// }
```
Comment (nathan-jenan-rancher) [inline pkg/api/server/managementstored/setup.go]: Delete commented out code

**Comment by erikwilson** [comment]
Update: Needs to retrieve ruleset information from cluster

**Comment by cjellick** [comment]
This stalled for various reasons. closing

**Pull request closed without merging**